### ISSUE TITLE
No "Manage Plug-ins..." available

### ISSUE THREAD

**Issue by markles2** [apparition47/MailTrackerBlocker#50]
Using OS 11.2.2, installed MailTrackerBlocker today, but there's no such option.

**Comment by OofyProsser** [comment]
The option did not show up for me, either, because I didn't actually install the plug-in, I just downloaded the package. After I installed it, the option showed up, I assume it was activated by the installation. I am on Catalina 10.15.7. 

Another possibility: Did you restart Mail?

**Comment by markles2** [comment]
I did install it, and restarted mail. Mail was quit before installation.

**Comment by OofyProsser** [comment]
Sorry, that's all I got. Someone more skilled than I will have to help.

Wait, one more idea in passing: how abut rebooting the Mac? Just a guess, though

**Comment by OofyProsser** [comment]
OK, just one more idea... I had to install it twice to get it to show up correctly.

**Comment by markles2** [comment]
Alas, no. Rebooted and reinstalled.  I suspect, instead, it's something basic.  Like: Is it supposed to just drop in a file somewhere?  Because I can't find it.  I just created a new folder, where there was none, at (user)/Library/Application Support/Mail/Plug-ins.

**Comment by OofyProsser** [comment]
The installer will put the various pieces where they belong. None of which go in /Mail/Plug-ins.

The first time I downloaded the file to the desktop then double-clicked on that, but that didn’t do a complete job of installing so I tried again. What worked best for me was to go to the website, click the download icon, then just let the Installer do its work.

**Comment by Mulder-X1** [comment]
> The option did not show up for me, either, because I didn't actually install the plug-in, I just downloaded the package. After I installed it, the option showed up, I assume it was activated by the installation. I am on Catalina 10.15.7.
> 
> Another possibility: Did you restart Mail?

MailTrackerBlocker doesn't work in macOS 10.11 or higher because at least through 10.13.6, there has never been a Manage Plug--ins option in the General preferences section. That's only been there in the most recent versions of macOS. MailTrackerBlocker isn't a plug-in either: it's a Mail Bundle that should appear separately in Mail Preferences like other bundles, but it doesn't.

I don't think the author of this has tested it in macOS 10.11 or higher; just the most recent version.

**Comment by markles2** [comment]
Ah! That explains it.  Thanks.

**Comment by apparition47** [comment]
@Mulder-X1 I usually test on 10.14, 10.15 and 11.2

@markles2 The "Manage Plug-ins" button is normally hidden by default. My installation pkg is supposed enable this but apparently it hasn't been working for a lot of people.

Could you let me know what the output of this is? 

```bash
sudo defaults read "/Library/Preferences/com.apple.mail" EnableBundles
defaults read com.apple.mail EnableBundles
defaults read com.apple.mail BundleCompatibilityVersion
```

**Comment by Mulder-X1** [comment]
> On Mar 2, 2021, at 7:57 PM, Aaron Lee <notifications@github.com> wrote:
> 
> @Mulder-X1 <https://github.com/Mulder-X1> I usually test on 10.14, 10.15 and 11.2
> @markles2 <https://github.com/markles2> The "Manage Plug-ins" button is normally hidden by default. My installation pkg is supposed enable this but apparently it hasn't been working for a lot of people.
> 
> Could you let me know what the output of this is?
> 
> sudo defaults read "/Library/Preferences/com.apple.mail” EnableBundles
1
> defaults read com.apple.mail EnableBundles
YES
> defaults read com.apple.mail BundleCompatibilityVersion
The domain/default pair of (/Users/Mulder/Library/Containers/com.apple.mail/Data/Library/Preferences/com.apple.mail, BundleCompatibilityVersion) does not exist

**Comment by apparition47** [comment]
@Mulder-X1 Are you having the same issue as @markles2 ?

**Comment by Mrb55-oregon** [comment]
Yes, I'm having the same issue, there's no Manage Plug-ins option available. I'm running 10.13.6
Thanks, Mark

**Comment by markles2** [comment]
Basically similar results as Multer-X1:
1
1
… does not exist.

I wonder if that last one, the non-existence of the folder/file is the problem.

FYI I'm running OS 11.2.2 (Big Sur)

**Comment by apparition47** [comment]
>I wonder if that last one, the non-existence of the folder/file is the problem.

Apparently not since I have the exact same output as @Mulder-X1 .


```bash
sudo defaults write "/Library/Preferences/com.apple.mail" EnableBundles 1
defaults write com.apple.mail EnableBundles -bool true
```

These commands (which are executed by the installer) are supposed to enable `Manage Plug-ins...` but for whatever reason, it's not working on your setups. Still investigating.

**Comment by apparition47** [comment]
ℹ️ (EDIT: 0.3.16 has been released. Let's work with that as a baseline.)

>I wonder if that last one, the non-existence of the folder/file is the problem.

Apparently not since I have the exact same output as @Mulder-X1 .


These commands (which are executed by the installer) are supposed to enable Manage Plug-ins.... Maybe try give it a try:

```bash
sudo defaults write "/Library/Preferences/com.apple.mail" EnableBundles 1
defaults write com.apple.mail EnableBundles -bool true
defaults write com.apple.mail BundleCompatibilityVersion 4
```

**Comment by Mulder-X1** [comment]
Yes, I am. I’m running macOS 10.13.6 and I’ve tried installing it via the package, and via Homebrew and t he result is the same. But currently, in 10.13.6 V5 is the current version for bundle compatibility.

**Comment by OofyProsser** [comment]
FWIW, I get the same results as Markles2 (1, 1, does not exist) yet my installation is working fine. Catalina 10.15.7.

**Comment by ronewby** [comment]
Additional result...In High Sierra, ran installation and got no "Manage Plug-ins" in Mail preferences, but the "x" button is working fine and identifying trackers.

**Comment by HandyMac** [comment]
Read about MailTrackerBlocker at Daring Fireball; sounded good, so I downloaded it. Running macOS 10.12.6 Sierra on my MacBook Pro 10,1. I ran the Installer, it seemed to work okay. Opened Mail, but there was no [Manage Plug-Ins] button at the bottom of Preferences > General as pictured. Did a search and found it had been installed: /Library/Mail/Bundles/MailTrackerBlocker.mailbundle. 

So went to my other MacBook Pro (11,4), started in Sierra, did the same there, same result: bundle installed, but no way to activate it. Started in Catalina, opened Mail, saw no [Manage Plug-Ins] button at the bottom of Preferences > General, but there was space in the window for something. So installed MailTrackerBlocker, and now the [Manage Plug-Ins] button appeared. Did a search, found MailTrackerBlocker.mailbundle installed as in Sierra. So it works in Catalina, though the button doesn't show unless a Plug-In is installed. But it doesn't seem to work in Sierra. 

Tried the three Terminal commands above in Sierra, still no [Manage Plug-Ins] button in Mail Preferences, *but* now there's space for it in the bottom of the window, where before the "Encrypted Messages" checkbox line was right down at the bottom of the window. 

And the little X button now appears with each message opened in Mail. So it's working, even without being enabled. 

So just for grins, since the download button says "macOS 10.11+", I tried it in 10.11.6 El Capitan on my MacBook. But the Installer balked: "MailTrackerBlocker.pkg can't be installed because its digital signature is invalid. The package may have been corrupted or tampered with. Get a new copy of the package and try again."

So I went to the Web page and downloaded it again. Same result. So it seems it can't be installed in 10.11. 

Anyway, thought you'd like to know.

**Comment by apparition47** [comment]
@ronewby @OofyProsser @HandyMac For those who are reporting it working without manually enabling it after installation, do you have SIP disabled?

>So it seems it can't be installed in 10.11.

Addressed in #53.

**Comment by ronewby** [comment]
SIP is enabled.

**Comment by apparition47** [comment]
Managed to provision a 10.13.6 system. Just to confirm what @Mulder-X1 said, MailTrackerBlocker will be automatically enabled on **10.13 and older**. Just look for the `ⓧ` icon to double check. I'll clarify this in the docs.

Still not sure why "Manage Plugins" isn't showing for certain Big Sur users like @markles2 yet. Still investigating this case.

**Comment by HandyMac** [comment]
> @ronewby @OofyProsser @HandyMac For those who are reporting it working without manually enabling it after installation, do you have SIP disabled?
> 
SIP is not disabled; didn't even think of it. 

> MailTrackerBlocker will be automatically enabled on 10.13 and older. Just look for the `ⓧ` icon to double check.

Yes. Just tried in 10.12 Sierra again (another setup where it hadn't been installed before). After running the installer, the ⓧ is present in Mail's message window, though the [Manage Plug-Ins] button does not appear in Mail's Preferences. Tried the three Terminal commands, one at a time, but the [Manage Plug-Ins] button did not appear. And this time Mail's Preferences window did not grow extra space at the bottom. Went back to the two Sierra setups where I'd installed MailTrackerBlocker last night; there is now no extra space in the Mail Preferences window in either (don't remember in which one that happened).

**Comment by Mulder-X1** [comment]
MailTrackerBlocker isn’t configured on my10.13.6 system. Doesn’t matter if SIP is enabled or disabled — the result is still the same.

**Comment by apparition47** [comment]
@HandyMac If you're seeing the `ⓧ` icon in the Mail message window beside the subject line and under the date, then you're ready to go.

<img width="789" alt="macOS_10_14_2" src="https://user-images.githubusercontent.com/3298414/110137810-989ffe80-7e14-11eb-9102-29553cbb4855.png">

**Comment by HandyMac** [comment]
> MailTrackerBlocker isn’t configured on my10.13.6 system. Doesn’t matter if SIP is enabled or disabled — the result is still the same.

SIP doesn't matter; it doesn't prevent installing MailTrackerBlocker. Go to /Library/Mail/Bundles/: is MailTrackerBlocker.mailbundle there? If it is, it should be working. If not, run the Installer again. 

Open a message in Mail, look to the right; the ⓧ is small and unobtrusive. Here it is in my 10.13.6 Mail: 

<img width="212" alt="highsierra" src="https://user-images.githubusercontent.com/31111465/110138003-ba56a280-7d8e-11eb-9d29-c1b402542f7f.png">

**Comment by Mulder-X1** [comment]
MailTrackerBlocker is definitely installed, and now that I know what to look for, I see a small blue ⓧ on the far right of a message, underneath the time of arrival. Still nothing in Mail preferences.

**Comment by OofyProsser** [comment]
I don’t recall seeing this mentioned, so I will point it out just in case anyone else missed it, too:

If you click on the blue ⓧ, MTB will tell you how many trackers were blocked and who they were from. The white ⓧ indicates that MTB did not find anything to block.

Jerry

**Comment by apparition47** [comment]
@Mulder-X1 @OofyProsser Trying to keep the plugin unobtrusive so that's probably why you missed missed it. Others were confused about how it works too (#33). Probably reinforces the need for better documentation. Anyway, sounds like the issue is solved in your cases.

@markles2 Since you're running Big Sur, you should be the one to see "Manage Plug-ins". Are you still having this issue on 0.3.17?

**Comment by markles2** [comment]
No.  I still do not see (or have?) the "Manage Plug-ins" option in Mail>Preferences>General after having installed 0.3.17 over Mac OS 11.2.2 ("Big Sur").

**Comment by HandyMac** [comment]
Which is the latest version of MailTrackerBlocker? I thought I would install it in Big Sur, to provide you more than one example. So I downloaded it fresh from the Web page. And found one I'd downloaded earlier today is actually a "later" version. The one I just downloaded is 127,427 bytes, Created March 6, 2021 at 6:02 PM, Modified March 6, 2021 at 6:02 PM. (These two times seem to be when I downloaded it just now here in MST.) The one I downloaded this morning from one of your links is 127,471 bytes, Created March 6, 2021 at 5:28 AM, Modified March 6, 2021 at 9:10 PM. (How it could have been modified ~3 hours in the future (from now) is a mystery to me.) Which one should I install in Big Sur?
